st/clitests/awss3api_spec.py

```diff
@@ -1841,6 +1853,335 @@ def get_md5_sum(fname):
+print(f"Response: {list_versions_response} Versions received: {object_versions}")
+check = all(item in object_versions for item in expected_versions_list)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+
+# Step 6: Test delimiter which is not present
+# Command:= aws s3api list-object-versions --bucket <bucket> --delimiter "="
+# Expected output:
+#   asdf, bo0, boo/0...boo/1, boo#, boo+, fo0, foo/0...foo/2, foo#123
+#   foo+123, qua0, quax/0...quax/3, quax#, quax+
+expected_versions_list = ['asdf', 'bo0', 'boo/0', 'boo/1', 'boo#', 'boo+',
+    'fo0', 'foo/0', 'foo/1', 'foo/2', 'foo#123', 'foo+123', 'qua0', 'quax/0',
+    'quax/1', 'quax/2', 'quax/3', 'quax#', 'quax+']
+delimiter = "="
+list_versions_options = {"delimiter":delimiter}
+result = AwsTest(('Aws list object versions using delimiter %s') % (delimiter))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get versions from the response
+object_versions = list_versions_response["versions"]
+print(f"Response: {list_versions_response} Versions received: {object_versions}")
+check = all(item in object_versions for item in expected_versions_list)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+
+# Step 7: Test delimiter = "/"
+# Command:= aws s3api list-object-versions --bucket <bucket> --delimiter "/"
+# Expected output:
+#   asdf, bo0, boo#, boo+, fo0, foo#123, foo+123, qua0, quax#, quax+
+#   <Items under COMMONPREFIXES are as follows>
+#     boo/, foo/, quax/
+delimiter = "/"
+expected_versions_list = ['asdf', 'bo0', 'boo#', 'boo+', 'fo0', 'foo#123', 'foo+123', 'qua0', 'quax#', 'quax+']
+expected_common_prefixes = ['boo/', 'foo/', 'quax/']
+list_versions_options = {"delimiter":delimiter}
+result = AwsTest(('Aws list object versions using delimiter %s') % (delimiter))\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get keys from the response
+object_versions = list_versions_response["versions"]
+common_prefixes = list_versions_response["prefix"]
+print(f"Response: {list_versions_response} Versions received: {object_versions} Common prefixes received: {common_prefixes}")
+check = all(item in object_versions for item in expected_versions_list)
+if check is False:
+    assert False, "Failed to match expected versions in the list"
+check = all(item in common_prefixes for item in expected_common_prefixes)
+if check is False:
+    assert False, "Failed to match expected common prefixes in the list"
```

Comment:
Should these both be in the same order, like versions? In that case, just use list comparison, same as versions.

Reply:
Done